### DISCUSSION TITLE
Floating window too big.

### DISCUSSION THREAD

**Discussion by Samueru-sama** [i3/i3#5533] [Q&A]
I have an issue when I open the track properties window of deadbeef, it opens by default as a floating window, but it's size is very big for some reason, and it is the only floating window of the application that is affected by this issue, every other floating window from the same application doesn't show this issue. 

I opened an issue at the deadbeef repo and was told that it is likely an issue with i3, since it should open as a 400x400 window.

https://github.com/DeaDBeeF-Player/deadbeef/issues/2960#issuecomment-1599447012

Screenshots: 

Track properties: https://imgur.com/a/lfSLrlc

Other windows: https://imgur.com/a/QsFCwaM

**Comment by Samueru-sama** [comment]
I found that by setting: 

for_window [title="Track Properties$"] resize set 500 500

The window spawns with the right size, however when the window is called on monitor 1, and only monitor 1, the floating window spawns off the screen.

This only happens on monitor 1, Monitors 2 and 3 don't show this issue.

I also noticed that if I toggle between floating and tiling mode, the window becomes huge again when it goes back to floating mode.

**Comment by orestisfl** [comment]
Please follow the instructions at https://i3wm.org/docs/debugging.html and provide a log file as it would be helpful here

**Comment by Samueru-sama** [comment]
> Please follow the instructions at https://i3wm.org/docs/debugging.html and provide a log file as it would be helpful here

https://logs.i3wm.org/logs/6205506985132032.bz2

The window name that has the issue is called 'Track Properties'.

**Reply by orestisfl** [reply]
According to this log, the geometry of the window we get is 1440x855:

```
06/21/2023 01:09:12 PM - ../i3/src/manage.c:manage_window:277 - Initial geometry: (0, 0, 1440, 855)
```

**Reply by orestisfl** [reply]
I tried to reproduce this myself and I can't. I even installed `deadbeef-git` as you said in the other ticket. The window shows up as 400x400

```
$ xprop
_NET_WM_DESKTOP(CARDINAL) = 2
WM_STATE(WM_STATE):
		window state: Normal
		icon window: 0x0
I3_FLOATING_WINDOW(CARDINAL) = 1
_NET_WM_STATE(ATOM) = _NET_WM_STATE_SKIP_TASKBAR, _NET_WM_STATE_SKIP_PAGER
WM_HINTS(WM_HINTS):
		Client accepts input or input focus: True
		Initial state is Normal State.
		window id # of group leader: 0x3200001
XdndAware(ATOM) = BITMAP
_MOTIF_DRAG_RECEIVER_INFO(_MOTIF_DRAG_RECEIVER_INFO) = 0x6c, 0x0, 0x5, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x10, 0x0, 0x0, 0x0
WM_TRANSIENT_FOR(WINDOW): window id # 0x3200003
_NET_WM_SYNC_REQUEST_COUNTER(CARDINAL) = 52431012
_NET_WM_WINDOW_TYPE(ATOM) = _NET_WM_WINDOW_TYPE_NORMAL
_NET_WM_USER_TIME(CARDINAL) = 2944995
_NET_WM_USER_TIME_WINDOW(WINDOW): window id # 0x32008a3
WM_CLIENT_LEADER(WINDOW): window id # 0x3200001
_NET_WM_PID(CARDINAL) = 390768
WM_LOCALE_NAME(STRING) = "en_DK.UTF-8"
WM_CLIENT_MACHINE(STRING) = "desktop"
WM_NORMAL_HINTS(WM_SIZE_HINTS):
		program specified location: 0, 0
		program specified minimum size: 400 by 400
		window gravity: NorthWest
WM_PROTOCOLS(ATOM): protocols  WM_DELETE_WINDOW, WM_TAKE_FOCUS, _NET_WM_PING, _NET_WM_SYNC_REQUEST
WM_CLASS(STRING) = "deadbeef", "Deadbeef"
WM_ICON_NAME(STRING) = "Track Properties"
_NET_WM_ICON_NAME(UTF8_STRING) = "Track Properties"
WM_NAME(STRING) = "Track Properties"
_NET_WM_NAME(UTF8_STRING) = "Track Properties"
```

```
2023-06-24 09:36:52 - ../i3/src/manage.c:manage_window:277 - Initial geometry: (411, 335, 400, 400)
```

**Comment by Samueru-sama** [comment]


**Reply by orestisfl** [reply]
Thanks, these are useful data for debugging.

Since your i3 config is quite involved, can you please try to reproduce it with a minimal config? Perhaps you can start with the default i3 config and see if it's still reproducible. Also, try to reset the deadbeef config to get that out of picture as well, just temporarily rename the `~/.config/deadbeef` directory so that it falls back to the default.

Finally, small request, can you please respond in the same thread? It makes tracking conversation a little bit easier.
